### PR TITLE
HTCONDOR-1951: Improve htcondor dag status

### PR DESCRIPTION
**Pull request by ColeBollig**
-Improved output for htcondor dag status command
-Added new DAGMan classad attribute DAG_AdUpdateTime that is
 a timestamp of when the DAGMan job ad was last updated by the
 DAGMan process

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

File: src/condor_tools/htcondor_cli/dagman.py [hunk 1]
```diff
@@ -64,19 +104,44 @@ class Status(Verb):
...
 
+    def multi(self, num, past=False):
+        if num > 1 and not past:
+            return f"{num:3.0f} are"
+        elif num > 1 and past:
+            return f"{num:3.0f} have"
+        elif not past:
+            return f"{num:3.0f} is"
+        else:
+            return f"{num:3.0f} has"
```
Comment (Todd-L-Miller) [inline src/condor_tools/htcondor_cli/dagman.py:115]: I approve of this attention to good grammar.

**Review by Todd-L-Miller — COMMENTED** [review]
In general, I would like to see a few blank lines in places where one logical section ends and another begins (before lines 201 and 242 come to mind), and it might also be nice to invert the sense of the test on line 202 (noping out early) to save a level of indent.  Other than that, although parts of the code are a bit repetitive, I fiddled around for a little bit and couldn't come up with anything much cleaner.

Fixing the time display is the only blocker here.

File: src/condor_tools/htcondor_cli/dagman.py (hunk 1 above)
Comment (ColeBollig) [reply src/condor_tools/htcondor_cli/dagman.py:115]: Thanks :)

File: src/condor_tools/htcondor_cli/dagman.py [hunk 2]
```diff
@@ -64,19 +104,44 @@ class Status(Verb):
...
         dag = None
-        dag_status = "IDLE"
+        job_status = { 1:"idle", 2:"running", 3:"removed from the queue", 4:"is completed", 5:"held", 6:"transfering output", 7:"suspended" }
+        dagman_status = { 0:"is running normally.",
+                          1:"had an error occur.",
+                          2:"had at least one node fail.",
+                          3:"is aborting because it recieved a DAG abort value.",
+                          4:"has recieved a signal to be remove.",
+                          5:"has detected a cycle.",
+                          6:"has been halted."}
```
[the remark this answers is not in the document]
Comment (ColeBollig) [reply src/condor_tools/htcondor_cli/dagman.py:126]: Currently there is a JobStatus global variable available for the entirety of the htcondor_cli, but this is just filled with all capital lettered status names so I prefer my personal wording in job_status for the outputted messages. Whether or not this more verbose version should be higher level is open for discussion, but not needed currently so I will leave job_status here for now.

 As for the dag_status the only verbs for the DAG noun currently are SUBMIT & STATUS, and SUBMIT doesn't need the dag_status at all. Plus, this table is quite verbose for general output. I will leave as is for now to give thought about where, what, and how to have this table of information.

File: src/condor_tools/htcondor_cli/dagman.py [hunk 3]
```diff
@@ -16,6 +19,43 @@
...
+    days  = math.trunc(time / (60 * 60 * 24))
+    hours = round(time / (60 * 60)) % 24
+    mins  = round(time / 60) % 60
+    secs  = time % 60
+    days_display = ""
+    if days == 1:
+        days_display = "1 day "
+    elif days > 1:
+        days_display = f"{days} days "
+    return f"{days_display}{hours:02.0f}:{mins:02.0f}:{secs:02.0f}"
```
[the remark this answers is not in the document]
Comment (ColeBollig) [reply src/condor_tools/htcondor_cli/dagman.py]: I decided on just importing the eventlog function that is similar for now. I agree that a lot of these bits should be in some separate shared module. We should have a discussion about what and where common helper things should exist.

**Review by Todd-L-Miller — APPROVED** [review]
No regression test, but I'm not sure that makes sense for a PR like this.

Otherwise looks good.  Approved pending a successful test run.

**Pull request merged**